### PR TITLE
Easy way to extract values from foreman api search

### PR DESCRIPTION
**Pull request by scitechfh**
With puppet 3.x without future parser it difficult access foreman search results. This parser function converts easily any search result to an array.

### CODE REVIEW

File: lib/puppet/parser/functions/foreman_extract_from_search.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#
+# The foreman_extract_from_search() parser takes a hash value with query results and
+# a second parameter as key to extract
+#
+# To use foreman_extract_from_search(), apply results of foreman() and a key.
+#
+# $hosts = foreman_extract_from_search(foreman($f),'name')
+#
+
+require "net/http"
```
Comment (ekohl) [inline lib/puppet/parser/functions/foreman_extract_from_search.rb]: Are these requires actually used?

Comment (scitechfh) [inline lib/puppet/parser/functions/foreman_extract_from_search.rb]: No. This is an copy and paste error.

**Comment by ekohl** [comment]
Looks like it makes sense, but I'd like @mmoll or @domcleal to have a look.

**Comment by domcleal** [comment]
I guess it's needed now because the default API version switched to v2 in Foreman 1.9, so the foreman() function now returns the pagination and results data instead of just the results.  We could change foreman() to always return results directly, but I guess that could break manifests just as badly.

**Comment by ekohl** [comment]
@domcleal and a parameter in foreman() to return just the results?

**Comment by domcleal** [comment]
Oh, I see this is doing something further.  Given a response like this:

    {"results": [ {"name":"foo"}, {"name":"bar"} ]}

It's extracting a key from the hashes inside the results list - a bit like a map() function.

So probably best to ignore my comment above - it may be useful, but it's separate.

**Comment by scitechfh** [comment]
Whats about add an result filter attribute to foreman()?

**Comment by domcleal** [comment]
That'd be fine by me too.

**Comment by scitechfh** [comment]
I have removed my old function and added an filter function to foreman(). It's possible convert the result into an array or an hash with limited key or renamed keys.

**Comment by domcleal** [comment]
Works great, merged as a41fce0.  Thanks @scitechfh and sorry for the delay.

**Pull request closed without merging**